test/absinthe/middleware/dataloader_test.exs

```diff
@@ -158,11 +159,57 @@ defmodule Absinthe.Middleware.DataloaderTest do
       ]
     }
 
+    # Get test PID and clear mailbox
+    self = self()
+    flush_process_mailbox()
+
+    :ok =
+      :telemetry.attach_many(
+        "#{test}",
+        [
+          [:absinthe, :plugin, :before_resolution, :start],
+          [:absinthe, :plugin, :before_resolution, :stop],
+          [:absinthe, :plugin, :after_resolution, :start],
+          [:absinthe, :plugin, :after_resolution, :stop]
+        ],
+        fn name, measurements, metadata, _ ->
+          send(self, {:telemetry_event, name, measurements, metadata})
+        end,
+        nil
+      )
+
     assert {:ok, %{data: data}} = Absinthe.run(doc, DefaultSchema)
     assert expected_data == data
 
     assert_receive(:loading)
     refute_receive(:loading)
+
+    # This test emits a total of 24 events, but there we are only validating the
+    # dataloader events in this test
+    results =
+      Enum.reduce(1..24, %{dataloader_starts: 0, dataloader_stops: 0}, fn _, acc ->
```

Comment:
I'm unclear why the `reduce` is needed, the other telemetry tests managed to work just fine with `assert_receive`...

https://github.com/absinthe-graphql/absinthe/blob/7821d278b8c511fbae752f5f68bd94195a9c2607/test/absinthe/integration/execution/telemetry_test.exs#L62

Reply:
The reason for the reduce is so that I can count only the specific events that I want to fetch. Since the telemetry events emit on all plugins, I was also getting Async and Batch events. But only wanted to assert on the Dataloader specific events.